Controller/Adminhtml/Menu/Save.php

```diff
@@ -249,16 +260,20 @@ private function validateProductNode(array $node)
     }
 
     /**
-     * Returns menu model based on the Request (requested with `id` or fresh instance)
+     * Returns menu model based on the Request (requested with `menu_id` or fresh instance)
      *
      * @return Menu
      */
     private function getCurrentMenu(): Menu
     {
-        $menuId = $this->getRequest()->getParam('id');
+        $menuId = (int) $this->getRequest()->getParam(self::ID);
 
         if ($menuId) {
-            return $this->menuRepository->getById($menuId);
+            try {
+                return $this->menuRepository->getById($menuId);
+            } catch (NoSuchEntityException $exception) {
+                return $this->menuFactory->create();
```

Comment:
are you sure?

Reply:
Why not? If there is no menu id, it creates a new menu model for saving data.